### PR TITLE
Support TLS in Elasticsearch cron jobs

### LINKED ISSUE #471: Unable to connect to the ES when using K8s secrets
HI,

I have setup the ES cleaner cronjob for an externally hosted Elastic search cluster, and i am trying to use secrets stored in kubernetes to connect to the ES cluster.

Below is my yaml
```apiVersion: batch/v1beta1
kind: CronJob
metadata:
  name: es-jaeger-cleaner
  namespace: monitoring
  labels:
    app: es-jaeger-cleaner
    env: staging
spec:
  # every 1 PM UTC-0
  schedule: "0 13 * * *"
  jobTemplate:
    metadata:
      labels:
        app: es-jaeger-cleaner
        env: staging
    spec:
      template:
        metadata:
          labels:
            app: es-jaeger-cleaner
            env: staging
        spec:
          containers:
          - name: es-jaeger-cleaner
            image: jaegertracing/jaeger-es-index-cleaner:latest
            # clean up ES data indices older than 7 days from now
            args: ["7", "<https://<esHOST:port>"]
            env:
            - name: TIMEOUT
              value: "300"
            - name: ES_USERNAME
              value: "<username>"
            - name: ES_PASSWORD
              valueFrom:
                secretKeyRef:
                  key: ES_PASSWORD
                  name: <secret_name>
          restartPolicy: OnFailure
```

And it somehow does not work and i get the below error

```
File "/es-index-cleaner/esCleaner.py", line 31, in main
    client = elasticsearch.Elasticsearch(sys.argv[2:], http_auth=(username, password))
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/client/__init__.py", line 206, in __init__
    self.transport = transport_class(_normalize_hosts(hosts), **kwargs)
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/transport.py", line 127, in __init__
    self.set_connections(hosts)
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/transport.py", line 177, in set_connections
    connections = list(zip(connections, hosts))
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/transport.py", line 173, in _create_connection
    return self.connection_class(**kwargs)
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/connection/http_urllib3.py", line 106, in __init__
    self.headers.update(urllib3.make_headers(basic_auth=http_auth))
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/request.py", line 65, in make_headers
    b64encode(b(basic_auth)).decode('utf-8')
  File "/usr/local/lib/python3.7/site-packages/urllib3/packages/six.py", line 620, in b
    return s.encode("latin-1")
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 9-13: ordinal not in range(256)
```

PS - If i hardcode the username and password, it works fine.
Please let me know if i am doing something wrong

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #471
and part of #592 

docs PR https://github.com/jaegertracing/documentation/pull/324

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: pkg/cronjob/es_index_cleaner.go [hunk 1]
```diff
@@ -83,19 +84,21 @@ func CreateEsIndexCleaner(jaeger *v1.Jaeger) *batchv1beta1.CronJob {
...
-									Name:      name,
-									Image:     jaeger.Spec.Storage.EsIndexCleaner.Image,
-									Args:      []string{strconv.Itoa(*jaeger.Spec.Storage.EsIndexCleaner.NumberOfDays), esUrls},
-									Env:       envs,
-									EnvFrom:   envFromSource,
-									Resources: commonSpec.Resources,
+									Name:         name,
+									Image:        jaeger.Spec.Storage.EsIndexCleaner.Image,
+									Args:         []string{strconv.Itoa(*jaeger.Spec.Storage.EsIndexCleaner.NumberOfDays), esUrls},
+									Env:          envs,
```
Comment (objectiser) [inline pkg/cronjob/es_index_cleaner.go]: Should this use `util.RemoveEmptyVars(envs)` as in rollover?

File: pkg/cronjob/es_rollover_test.go [hunk 2]
```diff
@@ -71,17 +72,31 @@ func TestEnvVars(t *testing.T) {
...
+				{Name: "ES_TLS_CERT", Value: "/etc/cert"},
+				{Name: "ES_TLS_KEY", Value: "/etc/key"},
+				{Name: "ES_TLS_SKIP_HOST_VERIFY", Value: "true"},
+			},
 		},
 	}
 	for _, test := range tests {
-		assert.Equal(t, test.expected, esScriptEnvVars(test.opts))
+		fmt.Println(EsScriptEnvVars(test.opts))
+		fmt.Println(EsScriptEnvVars(test.opts))
```
Comment (objectiser) [inline pkg/cronjob/es_rollover_test.go]: The two lines above should be removed.

File: pkg/storage/elasticsearch_dependencies.go [hunk 3]
```diff
@@ -53,22 +74,17 @@ func elasticsearchDependencies(jaeger *v1.Jaeger) []batchv1.Job {
...
-		envs = append(envs, corev1.EnvVar{Name: "REPLICAS", Value: val})
-	}
-	if val, ok := spec.Map()["es.username"]; ok {
-		envs = append(envs, corev1.EnvVar{Name: "ES_USERNAME", Value: val})
-	}
-	if val, ok := spec.Map()["es.password"]; ok {
-		envs = append(envs, corev1.EnvVar{Name: "ES_PASSWORD", Value: val})
+func envVars(opts v1.Options) []corev1.EnvVar {
+	var envs = cronjob.EsScriptEnvVars(opts)
+	//envs := []corev1.EnvVar{}
```
Comment (objectiser) [inline pkg/storage/elasticsearch_dependencies.go]: Can this be removed?

File: pkg/storage/elasticsearch_dependencies.go [hunk 4]
```diff
@@ -53,22 +74,17 @@ func elasticsearchDependencies(jaeger *v1.Jaeger) []batchv1.Job {
...
+	options := opts.Map()
+	for k, v := range map[string]string{
+		"es.num-shards":   "SHARDS",
```
Comment (objectiser) [inline pkg/storage/elasticsearch_dependencies.go]: Why are the other options/env vars not copied? index prefix, username and password?

I see from the test they are still env vars, so what changed?

Comment (pavolloffay) [reply pkg/storage/elasticsearch_dependencies.go]: They are copied in a function from `cronjob` package just right above. `var envs = cronjob.EsScriptEnvVars(opts)`

**Pull request merged**